lib/puppet/util/servicenow.rb

```diff
@@ -71,13 +71,18 @@ def settings(settings_file = Puppet[:confdir] + '/servicenow_reporting.yaml')
   end
   module_function :settings
 
-  def do_snow_request(uri, http_verb, body, user: nil, password: nil, oauth_token: nil)
+  def do_snow_request(uri, http_verb, body, user: nil, password: nil, oauth_token: nil, skip_cert_check: false)
     uri = URI.parse(uri)
+    verify_mode = skip_cert_check ? OpenSSL::SSL::VERIFY_NONE : OpenSSL::SSL::VERIFY_PEER
+
+    opts = {
+      use_ssl: uri.scheme == 'https',
+      verify_mode: verify_mode,
```

Comment:
Does `Net::HTTP` ignore the `verify_mode` parameter when not using https?

Reply:
Yes, I double checked with some manual testing, and it does ignore unnecessary SSL mode settings that aren't needed when hitting an HTTP end point.